lib/ansible/modules/cloud/hcloud/hcloud_network.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_network
+
+short_description: Create and manage cloud Networks on the Hetzner Cloud.
+
+version_added: "2.9"
+
+description:
+    - Create, update and manage cloud Networks on the Hetzner Cloud.
+    - You need at least hcloud-python 1.3.0
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    id:
+        description:
+            - The ID of the Hetzner Cloud Networks to manage.
+            - Only required if no Network I(name) is given
```

Comment:
One thing that would be noteworthy to users: if the user specifies both `id` and `name` during creation, the network will not necessarily have the ID the user specified.

Reply:
I'm not really sure if we need to handle this, because we don't handle this in the other modules as well.

Reply:
It would mostly reduce the potential number of support queries when users wonder why something is not working from their POV ;-) Feel free to ignore this; I'm mostly pointing out what I would be wondering about if I would be a user of the module :-)